lib/bolt/inventory/group2.rb

```diff
@@ -253,37 +247,30 @@ def data_merge(data1, data2)
         }
       end
 
-      private def fetch_value(data, key, type)
-        value = data.fetch(key, type.new)
-        unless value.is_a?(type)
-          raise ValidationError.new("Expected #{key} to be of type #{type}, not #{value.class}", @name)
-        end
-        value
-      end
-
-      def resolve_aliases(aliases, target_names)
-        @name_or_alias.each do |name_or_alias|
-          # If an alias is found, insert the name into this group. Otherwise use the name as a new target's uri.
-          if target_names.include?(name_or_alias)
-            @targets[name_or_alias] = { 'name' => name_or_alias }
-          elsif (target_name = aliases[name_or_alias])
-            if @targets.include?(target_name)
-              @logger.warn("Ignoring duplicate target in #{@name}: #{target_name}")
+      def resolve_string_targets(aliases, known_targets)
+        @string_targets.each do |string_target|
+          # If this is the name of a target defined elsewhere, then insert the
+          # target into this group as just a name. Otherwise, add a new target
+          # with the string as the URI.
+          if known_targets.include?(string_target)
+            @unresolved_targets[string_target] = { 'name' => string_target }
+          # If this is an alias for an existing target, then add it to this group
+          elsif (canonical_name = aliases[string_target])
+            if local_targets.include?(canonical_name)
+              @logger.warn("Ignoring duplicate target in #{@name}: #{canonical_name}")
```

Comment:
Do we need this warning message? If there is no data on the target nothing meaningful is being ignored right?

Reply:
It was already there, so I left it. I don't think it's especially helpful though.